jetty-util/src/test/java/org/eclipse/jetty/util/PoolTest.java

```diff
@@ -331,34 +347,44 @@ public void testMaxUsageCount(int cacheSize)
     }
 
     @ParameterizedTest
-    @MethodSource(value = "cacheSize")
-    public void testMaxMultiplex(int cacheSize)
+    @MethodSource(value = "strategy")
+    public void testMaxMultiplex(Factory factory)
     {
-        Pool<String> pool = new Pool<>(2, cacheSize);
+        Pool<String> pool = factory.getPool(2);
         pool.setMaxMultiplex(3);
-        pool.reserve(-1).enable("aaa", false);
-        pool.reserve(-1).enable("bbb", false);
 
-        Pool<String>.Entry e1 = pool.acquire();
-        Pool<String>.Entry e2 = pool.acquire();
-        Pool<String>.Entry e3 = pool.acquire();
-        Pool<String>.Entry e4 = pool.acquire();
-        assertThat(e1.getPooled(), equalTo("aaa"));
-        assertThat(e1, sameInstance(e2));
-        assertThat(e1, sameInstance(e3));
-        assertThat(e4.getPooled(), equalTo("bbb"));
-        assertThat(pool.release(e1), is(true));
-        Pool<String>.Entry e5 = pool.acquire();
-        assertThat(e2, sameInstance(e5));
-        Pool<String>.Entry e6 = pool.acquire();
-        assertThat(e4, sameInstance(e6));
+        Map<String, AtomicInteger> counts = new HashMap<>();
+        AtomicInteger a = new AtomicInteger();
+        AtomicInteger b = new AtomicInteger();
+        counts.put("a", a);
+        counts.put("b", b);
+        pool.reserve(-1).enable("a", false);
+        pool.reserve(-1).enable("b", false);
+
+        counts.get(pool.acquire().getPooled()).incrementAndGet();
+        counts.get(pool.acquire().getPooled()).incrementAndGet();
+        counts.get(pool.acquire().getPooled()).incrementAndGet();
+        counts.get(pool.acquire().getPooled()).incrementAndGet();
+
+        assertThat(a.get(), greaterThan(0));
+        assertThat(a.get(), lessThanOrEqualTo(3));
+        assertThat(b.get(), greaterThan(0));
+        assertThat(b.get(), lessThanOrEqualTo(3));
+
+        counts.get(pool.acquire().getPooled()).incrementAndGet();
+        counts.get(pool.acquire().getPooled()).incrementAndGet();
+
+        assertThat(a.get(), is(3));
+        assertThat(b.get(), is(3));
+
+        assertNull(pool.acquire());
     }
 
     @ParameterizedTest
-    @MethodSource(value = "cacheSize")
-    public void testRemoveMultiplexed(int cacheSize)
+    @MethodSource(value = "strategy")
+    public void testRemoveMultiplexed(Factory factory)
     {
-        Pool<String> pool = new Pool<>(1, cacheSize);
+        Pool<String> pool = factory.getPool(1);
         pool.setMaxMultiplex(2);
         pool.reserve(-1).enable("aaa", false);
```

Comment:
This test contains an unused `e2` variable that I would instead add to the asserts, for example, to confirm that it is in use even after removing `e1`, or that `e2 == e1`, etc.